posts/2022/02/making-gpus-accessible-to-pydata-ecosystem-via-array-api.md

```diff
@@ -0,0 +1,497 @@
+the runtime of the demo code on CPU and GPU as a function of input array size.
+The x-axis of the plot is the image dimension, i.e the dimension of the original
+image after rescaling it by a constant factor along both dimensions. For this
+demo the rescaling factors were: 0.1, 0.2, 0.4, 0.6, 0.8 and 1.
+
+
+#### On AMD GPU:
+
+![NumPy cs CuPy AMD](/images/2022/02/numpy_vs_cupy_amd.png)
+
+
+#### On NVIDIA GPU:
+
+![NumPy cs CuPy NVIDIA](/images/2022/02/numpy_vs_cupy_nvidia.png)
+
+This was run on a NVIDIA TITAN RTX. The GPU vs. CPU plot is what you would
+expect: the computation is faster with CuPy (i.e. on GPU) compared to NumPy
+(i.e. on CPU) for larger image sizes.  The computation on GPU is slower than on
+CPU for image sizes less than **61 x 77**. This is due to the overhead of
+moving data to the device (GPU), which is significant compared to the time
+taken for actual computation. The larger the image, the longer the computation
+takes, and hence the less important the data transfer becomes.
+
+
+### Reproducing these results
+
+The code and instructions to run the analysis above can be found in this repository:
+[Quansight-Labs/array-api-gpu-demo](https://github.com/Quansight-Labs/array-api-gpu-demo).
+
+The above analysis required changes in the following libraries:
+
+- `cupy`
+- `scipy`
+- `scikit-learn`
+- `scikit-image`
+
+The changes are present in the `array-api-gpu-demo` branch of [@aktech](https://github.com/aktech)'s fork
+of the above projects.
+
+In the above mentioned repo, you can find the code with which we created Docker
+images for both NVIDIA & AMD platforms. The README also contains instructions
+for how to obtain the built Docker images from Docker Registry. Once you are
+inside the Docker container, running the demo is as simple as running:
+
+```bash
+python segmentation_performance.py
+```
+
+Running the ROCm container on a machine with an AMD GPU is done like this:
+
+```bash
+docker run -it --device=/dev/kfd --device=/dev/dri --security-opt seccomp=unconfined --group-add video ghcr.io/quansight-labs/array-api-gpu-demo-rocm:latest bash
+```
+
+
+## Next Steps
+
+At the moment we have only modified the functions we needed to be able to run
+our demo on GPU. There is of course a lot more functionality that can
+eventually be ported to support the Array API standard. The eventual goal of
+that is to make array-consuming libraries like SciPy, scikit-learn, and
+scikit-image fully compatible with multiple array libraries.
+
+
+## Acknowledgment
+
+This demo work was part of a larger project to make PyData libraries work with
+multiple array libraries and thereby run on GPUs. This is a significant
+undertaking, and this work would not have been possible without funding
+from AMD for a team at Quansight Labs.
+
+Special thanks to [Ivan Yashchuk](https://github.com/IvanYashchuk) and
+[Ralf Gommers](https://github.com/rgommers) for their guidance throughout the
+project.
```

Comment:
I appreciate the thought @aktech, but I'm thinking it may be preferred to leave this acknowledgement out. Other team members contributed as well to this work, and the main purpose of this section is funding acknowledgement. 

@IvanYashchuk WDYT?

Reply:
Let's keep only funding acknowledgement.